experimental/dds/tree2/src/test/feature-libraries/default-field-kinds/optionalChangeRebaser.spec.ts

```diff
@@ -137,29 +177,103 @@ function compose(changes: TaggedChange<OptionalChangeset>[]): OptionalChangeset
-	["SetA", OptionalChange.set("A", false)],
-	["SetB", OptionalChange.set("B", false)],
-	["SetUndefined", OptionalChange.set(undefined, false)],
-	["ChangeChild", OptionalChange.buildChildChange(TestChange.mint([], 1))],
-];
-deepFreeze(testChanges);
+type OptionalFieldTestState = FieldStateTree<string | undefined, OptionalChangeset>;
+
+/**
+ * See {@link ChildStateGenerator}
+ */
+const generateChildStates: ChildStateGenerator<string | undefined, OptionalChangeset> = function* (
+	state: OptionalFieldTestState,
+	tagFromIntention: (intention: number) => RevisionTag,
+	mintIntention: () => number,
+): Iterable<OptionalFieldTestState> {
+	const inputContext = getInputContext(state);
+	if (state.content !== undefined) {
+		const changeChildIntention = mintIntention();
+		yield {
+			content: state.content,
+			mostRecentEdit: {
+				changeset: tagChange(
+					OptionalChange.buildChildChange(
+						TestChange.mint(inputContext, changeChildIntention),
+					),
+					tagFromIntention(changeChildIntention),
+				),
+				intention: changeChildIntention,
+				description: `ChildChange${changeChildIntention}`,
+			},
+			parent: state,
+		};
+
+		const setUndefinedIntention = mintIntention();
+		yield {
+			content: undefined,
+			mostRecentEdit: {
+				changeset: tagChange(
+					OptionalChange.set(undefined, false),
+					tagFromIntention(setUndefinedIntention),
+				),
+				intention: setUndefinedIntention,
+				description: "Delete",
+			},
+			parent: state,
+		};
+	}
+
+	for (const value of ["A", "B"]) {
+		const setIntention = mintIntention();
+		// Using length of the input context guarantees set operations generated at different times also have different
+		// values, which should tend to be easier to debug
+		const newContents = `${value},${inputContext.length}`;
+		yield {
+			content: newContents,
+			mostRecentEdit: {
+				changeset: tagChange(
+					OptionalChange.set(newContents, state.content === undefined),
+					tagFromIntention(setIntention),
+				),
+				intention: setIntention,
+				description: `Set${value},${inputContext.length}`,
+			},
+			parent: state,
+		};
+	}
+
+	if (state.mostRecentEdit !== undefined) {
+		const undoIntention = mintIntention();
+		yield {
+			content: state.parent?.content,
+			mostRecentEdit: {
+				changeset: tagChange(
+					invert(state.mostRecentEdit.changeset),
+					tagFromIntention(undoIntention),
+				),
+				// TODO: Invert/TestChange has some funky logic around negative intentions being used
+				// for inverses. The handling of intention/revision here might need to be updated.
+				// e.g. Should this be -state.mostRecentEdit.intention?
+				intention: undoIntention,
+				description: `Undo:${state.mostRecentEdit.description}`,
+			},
+			parent: state,
+		};
+	}
+};
+
+/**
+ * Runs a suite of axiomatic tests which use combinations of single edits that are valid to apply from an initial state.
+ */
+function runSingleEditRebaseAxiomSuite(initialState: OptionalFieldTestState) {
```

Comment:
Consider making most of the new code in this file FieldKind-agnostic so it can used with other fields kinds and also so the code itself can be tested by running on `TestChange` which will error if driven to compute something invalid.

Reply:
I took a shot at doing this. It might not be exactly where we want to land long-term--see comment on `rebaseComposed`--but it does seem pretty reasonable to add analogous coverage for sequence-field and/or TestChange (which I'd like to do separately).